src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -129,26 +213,75 @@ private async Task<T> UpsertSystemObjectAsync<T>(T systemObject, string partitio
                 PartitionKey = new PartitionKey(partitionKey),
                 AccessCondition = eTagAccessCondition,
             };
-
             try
             {
                 ResourceResponse<Document> response = await _documentClient.Value.UpsertDocumentAsync(
+                   _collectionUri,
+                   systemObject,
+                   requestOptions,
+                   true,
+                   cancellationToken);
+                _logger.LogInformation("Request charge: {RequestCharge}, latency: {RequestLatency}", response.RequestCharge, response.RequestLatency);
+                return (dynamic)response.Resource;
+            }
+            catch (DocumentClientException dce)
+            {
+                _logger.LogError(dce, "Unhandled Document Client Exception");
+                throw;
+            }
+        }
+
+        private async Task<T> AddSystemObjectAsync<T>(T systemObject, string partitionKey, CancellationToken cancellationToken)
+           where T : class
+        {
+            EnsureArg.IsNotNull(systemObject, nameof(systemObject));
+            var eTagAccessCondition = new AccessCondition();
+
+            var requestOptions = new RequestOptions
+            {
+                PartitionKey = new PartitionKey(partitionKey),
+                AccessCondition = eTagAccessCondition,
+            };
+            try
+            {
+                ResourceResponse<Document> response = await _documentClient.Value.CreateDocumentAsync(
                     _collectionUri,
                     systemObject,
                     requestOptions,
                     true,
                     cancellationToken);
                 _logger.LogInformation("Request charge: {RequestCharge}, latency: {RequestLatency}", response.RequestCharge, response.RequestLatency);
-
-                return (T)(dynamic)response.Resource;
+                return (dynamic)response.Resource;
             }
             catch (DocumentClientException dce)
             {
-                if (dce.StatusCode == HttpStatusCode.TooManyRequests)
```

Comment:
Looks like the check for `HttpStatusCode.TooManyRequests` was accidentally removed. Also need to check this in the other queries, too.